src/evo/providertx.cpp

```diff
@@ -178,6 +178,13 @@ bool CheckNewAssetTx(const CTransaction &tx, const CBlockIndex *pindexPrev, CVal
         return state.DoS(100, false, REJECT_INVALID, "bad-assets-collateralAddress");
     }
 
+    if (assetTx.referenceHash.length() != 0){
+        if (assetTx.referenceHash.length() == 46){
+            if (assetTx.referenceHash.substr(0, 2) != "Qm")
+                return state.DoS(100, false, REJECT_INVALID, "bad-assets-referenceHash");
+        }
+        return state.DoS(100, false, REJECT_INVALID, "bad-assets-referenceHash");
+    }
```

Comment:
```suggestion
    if (assetTx.referenceHash.length() != 0)
        return state.DoS(100, false, REJECT_INVALID, "bad-assets-referenceHash");

    if (assetTx.referenceHash.length() == 46 && assetTx.referenceHash.substr(0, 2) != "Qm")
        return state.DoS(100, false, REJECT_INVALID, "bad-assets-referenceHash");
```

Reply:
this change will reject any referenceHash that has length greater than 0.
valid referenceHash can have length 0 or length 46 and start with "Qm"

Reply:
This is the equivalent. At line 189, any referenceHash that has a length greater than 0 will be rejected.

Reply:
yeah... I hesitated for a moment..
This is the absolute equivalent. At line 189, any referenceHash that has a length greater than 0 will be rejected.

Reply:
it looks like it passes with length 0 only. ((